### PR TITLE
[Android] Fix UI test 1851

### LINKED ISSUE #2255: Issue1851 missing preserve attribute for android, UI Test, and a Scroll View wouldn't hurt
- UI test to click the button and make sure it doesn't crash
- Preserve attribute on all the classes so it can run on Android
- Put the StackLayout in a ScrollView so on small devices one can see the group getting added and removed

### PR DESCRIPTION
**Pull request by adrianknight89**
### Description of Change ###

Fixed UI test 1851 and automated it with a ScrollView wrapping the StackLayout. Note that I'm unable to run the automated tests at the moment due to VS not recognizing NUnit references in my workspace.

### Issues Resolved ### 

- fixes #2255

### Platforms Affected ### 

- Android

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by rmarinho** [comment]
build -uitests

**Comment by rmarinho** [comment]
build --uitests

**Comment by rmarinho** [comment]
build failing 

```
 D:\a\1\s\Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\Issue1851.cs(76,4): error CS0103: The name 'RunningApp' does not exist in the current context [D:\a\1\s\Xamarin.Forms.Core.macOS.UITests\Xamarin.Forms.Core.macOS.UITests.csproj]
2018-09-24T11:21:45.1343007Z         
```

**Comment by adrianknight89** [comment]
@rmarinho Is this isolated to MacOS? I don't know if I need to do anything on my end.

Also, do you all plan on upgrading your tests to use NUnit 3? I think they are still using 2.

**Comment by rmarinho** [comment]
@adrianknight89 Xamarin.UITest only supports NUnit 2

The error is happening in all UITest projects.

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1851.cs [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
+#endif
+
 namespace Xamarin.Forms.Controls.TestCasesPages
 {
+#if UITEST
+	[Category(UITestCategories.ListView)]
+#endif
 	[Preserve (AllMembers=true)]
 	[Issue (IssueTracker.Github, 1851, "ObservableCollection in ListView gets Index out of range when removing item", PlatformAffected.Android)]
 	public class Issue1851 : ContentPage
```
Comment (PureWeen) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1851.cs]: @adrianknight89  changing the base class to *TestContentPage* and moving the code to OnInit should fix the build errors

Comment (adrianknight89) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1851.cs]: Done. Another thing I noticed is you have 18 issue tests under `Xamarin.Forms.Controls.TestCasesPages` whereas most everything belongs to the `Xamarin.Forms.Controls.Issues` namespace.

Comment (adrianknight89) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue1851.cs]: @StephaneDelcroix please verify the new commit. Thanks!

**Comment by samhouts** [comment]
@StephaneDelcroix @PureWeen Please re-review

**Comment by StephaneDelcroix** [comment]
build --uitests

**Comment by PureWeen** [comment]
failing UI tests aren't related to PR

**Pull request merged**